File: test/interface/golden/__tests__/file/File_spec.ml
let () =
  AesonSpec.goldenDirSpec
    File.decodePerson
    File.encodePerson
    "person"
    "golden/file/Person";

  AesonSpec.goldenDirSpec
    File.decodeAutomobile
    File.encodeAutomobile
    "automobile"
    "golden/file/Automobile";

  AesonSpec.goldenDirSpec
    File.decodeBusiness
    File.encodeBusiness
    "business"
    "golden/file/Business";

  AesonSpec.goldenDirSpec
    (File.decodeWrapper AesonSpec.decodeIntWithResult AesonSpec.decodeIntWithResult)
    (File.encodeWrapper Aeson.Encode.int Aeson.Encode.int)
    "wrapper"
    "golden/file/Wrapper";

  AesonSpec.goldenDirSpec
    File.decodeAutoDependingOnManual
    File.encodeAutoDependingOnManual
    "autoDependingOnManual"
    "golden/file/AutoDependingOnManual";

  AesonSpec.goldenDirSpec
    File.decodeNonGenericType
    File.encodeNonGenericType
    "nonGenericType"
    "golden/file/NonGenericType";

File: test/interface/golden/lib/js/__tests__/file/File_spec.js
'use strict';

var File      = require("../../file/File.js");
var AesonSpec = require("bs-aeson-spec/lib/js/src/AesonSpec.js");

AesonSpec.sampleGoldenAndServerSpec(File.decodePerson, File.encodePerson, "person", "http://localhost:8083/File/Person", "golden/file/Person");

AesonSpec.sampleGoldenAndServerSpec(File.decodeAutomobile, File.encodeAutomobile, "automobile", "http://localhost:8083/File/Automobile", "golden/file/Automobile");

AesonSpec.sampleGoldenAndServerSpec(File.decodeBusiness, File.encodeBusiness, "business", "http://localhost:8083/File/Business", "golden/file/Business");

AesonSpec.sampleGoldenAndServerSpec((function (param) {
        return File.decodeWrapper(AesonSpec.decodeIntWithResult, AesonSpec.decodeIntWithResult, param);
      }), (function (param) {
        return File.encodeWrapper((function (prim) {
                      return prim;
                    }), (function (prim) {
                      return prim;
                    }), param);
      }), "wrapper", "http://localhost:8083/File/Wrapper", "golden/file/Wrapper");

/*  Not a pure module */
